src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/WireHopperTest.java

```diff
@@ -0,0 +1,179 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.net.NetClient;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.net.NetServer;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.GradleTaskForRca;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.SymptomFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.metrics.Heap_Used;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.GenericFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.Node;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.util.RcaConsts;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.messages.DataMsg;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.messages.IntentMsg;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net.handler.PublishRequestHandler;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net.handler.SubscribeServerHandler;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ClusterDetailsEventProcessor;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(GradleTaskForRca.class)
+public class WireHopperTest {
+    private static final String NODE1 = "NODE1";
+    private static final String NODE2 = "NODE2";
+    private static final String LOCALHOST = "127.0.0.1";
+    private static final String HOST_NOT_IN_CLUSTER = "NOTINCLUSTER";
+    private static final String LOCUS = "data-node";
+    private static final long EVAL_INTERVAL_S = 5L;
+    private static final long TIMESTAMP = 66L;
+    private static final ExecutorService rejectingExecutor = new RejectingExecutor();
+
+    private static NetClient netClient;
+    private static NetServer netServer;
+    private static ExecutorService executorService;
+    private static ExecutorService netServerExecutor;
+    private static GRPCConnectionManager connectionManager;
+    private static SubscriptionManager subscriptionManager;
+    private static NodeStateManager nodeStateManager;
+    private static ReceivedFlowUnitStore receivedFlowUnitStore;
+    private static AtomicReference<ExecutorService> clientExecutor;
+    private static AtomicReference<ExecutorService> serverExecutor;
```

Comment:
Is there a reason all these have to be `static` ?

Reply:
I changed a few of them which didn't need to be static. The server startup stuff happens in the static @BeforeClass method